.release-notes/4182.md

```diff
@@ -0,0 +1,29 @@
+## Enhance checking for identity comparison with new objects
+
+An identity comparison between a new object and something else is always going to fail.
+
+This was correctly spotted by the compiler when using sugared constructor calls.
+
+Consider the following program:
+
+```pony
+class C
+
+actor Main
+  new create(env: Env) =>
+    env.out.print(if C is C then "C is C" else "C is NOT C" end)
+```
+
+This was going to fail during the compilation step with the message `identity comparison with a new object will always be false`.
+
+Nevertheless, the following code was compiled correctly:
```

Comment:
I would change this paragraph to say that the checking wasn't exhaustive and some cases weren't covered and then show your example.